### DISCUSSION TITLE
New to the Community, Looking for Advice on Contributing Effectively

### DISCUSSION THREAD

**Discussion by danilt2000** [dotnet/maui#24420] [Q&A] [answered]
Hi everyone,

I'm fairly new to this community and really excited to contribute to the framework. I've already started by trying to resolve some bug tickets. Occasionally, I can see potential solutions, but I'm concerned that some of my approaches might get rejected. On the other hand, there are also more complex bugs that I find challenging, and I'm not even sure where to begin.

Does anyone have advice on how I can make the most impactful contributions? Are there specific types of issues or areas where a beginner can be particularly helpful, or any tips on how to approach more difficult problems?

I appreciate any guidance you can offer!

**Comment by MartyIX** [comment, marked as answer]
Good question.

> I've already started by trying to resolve some bug tickets. Occasionally, I can see potential solutions, but I'm concerned that some of my approaches might get rejected.

Being fearful is not a good attitude. Just post a PR and possibly describe weak points. That's it. If it works, great. If not, you'll learn something new and slowly grow your skills. I've been learning the same way. Sometimes I got it right, sometimes I got it wrong. The important part is: You spend your time to improve something. That's what counts in my mind.

You can ping me on Discord if you want some help with contributing.

> Does anyone have advice on how I can make the most impactful contributions? Are there specific types of issues or areas where a beginner can be particularly helpful, or any tips on how to approach more difficult problems?

There are some issues marked with the good first issue label: https://github.com/dotnet/maui/issues?q=is%3Aopen+is%3Aissue+label%3A%22good+first+issue%22

Not sure about tips. Hard issues are hard. So it's often good to discuss them upfront to avoid wasting your time.

**Comment by danilt2000** [comment]
@MartyIX  Thank you for the advice! I'll definitely try to work on those issues and with this advice.
By the way, where can I find your Discord so I can reach out if I need some guidance? This is my https://discordapp.com/users/293977705815343105

**Reply by MartyIX** [reply]
https://aka.ms/dotnet-discord and look for "maui" channel. I tried to invite you directly but my Discord skills are not good enough :)

**Reply by danilt2000** [reply]
Thanks, I went. To be honest, I didn't even know maui have Discord channel.
